pkg/cri/resource-manager/control/rdt/rdt.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rdt
+
+import (
+	"fmt"
+
+	"github.com/intel/cri-resource-manager/pkg/cri/client"
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/cache"
+	"github.com/intel/cri-resource-manager/pkg/cri/resource-manager/control"
+	"github.com/intel/cri-resource-manager/pkg/rdt"
+	"github.com/intel/cri-resource-manager/pkg/utils"
+
+	"github.com/intel/cri-resource-manager/pkg/config"
+	logger "github.com/intel/cri-resource-manager/pkg/log"
+)
+
+const (
+	// RDTController is the name of the RDT controller.
+	RDTController = cache.RDT
+)
+
+// rdtctl encapsulates the runtime state of our RTD enforcment/controller.
+type rdtctl struct {
+	rdt   *rdt.Control // resctrl RDT control
+	cache cache.Cache  // resource manager cache
+}
+
+// Make sure rdtctl implements the control.Controller interface.
+var _ control.Controller = &rdtctl{}
+
+// Our logger instance.
+var log logger.Logger = logger.NewLogger(RDTController)
+
+// our RDT controller singleton instance.
+var singleton *rdtctl
+
+// getRDTController returns our singleton RDT controller instance.
+func getRDTController() control.Controller {
+	if singleton == nil {
+		singleton = &rdtctl{}
+	}
+	return singleton
+}
+
+// Start initializes the controller for enforcing decisions.
+func (ctl *rdtctl) Start(cache cache.Cache, client client.Client) error {
+	if ctl.rdt != nil {
+		return nil
+	}
+
+	rdtc, err := rdt.NewControl(opt.ResctrlPath)
+	if err != nil {
+		return rdtError("failed to create RDT controller: %v", err)
+	}
+
+	ctl.rdt = &rdtc
+	ctl.cache = cache
+
+	return nil
+}
+
+// Stop shuts down the controller.
+func (ctl *rdtctl) Stop() {
+}
```

Comment:
Just a mental note that maybe we should implement start/stop (or similar) in pkg/rdt, too(?) I.e. Stop() would delete all resctrl groups (managed by cri-rm) and thus return tasks within to the default/root group.

Reply:
Yes, that's something we need to figure out for rdt, and also for the block I/O controller once we start filling in the stubs.